federation/pkg/federation-controller/sync/controller.go

```diff
@@ -298,12 +299,13 @@ func (s *FederationSyncController) reconcile(namespacedName types.NamespacedName
 	if err != nil {
 		glog.Errorf("Error in retrieving %s from store: %v", kind, err)
 		s.deliver(namespacedName, 0, true)
-		return
+		return err
 	}
+
 	if !s.adapter.IsExpectedType(copiedObj) {
 		glog.Errorf("Object is not the expected type: %v", copiedObj)
 		s.deliver(namespacedName, 0, true)
-		return
+		return fmt.Errorf("Object is not the expected type: %v", copiedObj)
```

Comment:
Object -> object. error messages should be all lowercase, since we embed error messages